hedera-node/src/main/java/com/hedera/services/throttling/DeterministicThrottling.java

```diff
@@ -134,22 +121,32 @@ public boolean wasLastTxnGasThrottled() {
 	}
 
 	@Override
-	public void leakUnusedGasPreviouslyReserved(long value) {
+	public void leakUnusedGasPreviouslyReserved(TxnAccessor accessor, long value) {
+		if (accessor.throttleExempt()) {
+			return;
+		}
 		gasThrottle.leakUnusedGasPreviouslyReserved(value);
 	}
 
 	@Override
 	public boolean shouldThrottleQuery(HederaFunctionality queryFunction, Instant now, Query query) {
+		resetLastAllowedUse();
 		if (isGasThrottled(queryFunction) &&
 				dynamicProperties.shouldThrottleByGas() &&
 				(gasThrottle == null || !gasThrottle.allow(now, query.getContractCallLocal().getGas()))) {
+			reclaimLastAllowedUse();
```

Comment:
Is this needed? Seems we haven't yet had the opportunity to use capacity in anything but the `gas` bucket.